Common/Sources/Helper/CurrencyFormatter.swift

```diff
@@ -0,0 +1,236 @@
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import Foundation
+
+/// Transfer Amount Currency Formatter 
+public struct CurrencyFormatter {
+    private static let currencyData: CurrencyData? = {
+        if let path = Bundle(for: HyperwalletBundle.self).path(forResource: "Currency", ofType: "json"),
+            let data = NSData(contentsOfFile: path) as Data? {
+            return try? JSONDecoder().decode(CurrencyData.self, from: data)
+        }
+        return nil
+    }()
+    
+    /// Get Currency
+    /// - Parameter currencyCode: currency code
+    /// - Returns: instance of Currency
+    public static func getCurrency(for currencyCode: String) -> Currency? {
+        return currencyData?.data.first(where: { $0.currencyCode == currencyCode })
+    }
+    
+    /// Format double amount to currency string
+    /// - Parameters:
+    ///   - amount: amount
+    ///   - currencyCode: currency code
+    /// - Returns: a formatted currency string
+    public static func formatDoubleAmount(_ amount: Double, with currencyCode: String) -> String {
+        if let transferAmountCurrency = getCurrency(for: currencyCode) {
+            let formatter = NumberFormatter()
+            formatter.usesGroupingSeparator = true
+            formatter.maximumFractionDigits = transferAmountCurrency.decimals
+            formatter.minimumFractionDigits = transferAmountCurrency.decimals
+            formatter.numberStyle = .currency
+            formatter.currencyCode = currencyCode
+            formatter.locale = getLocaleIdentifer(for: currencyCode)
+            formatter.currencySymbol = ""
+
+            if let amount = formatter.string(from: NSNumber(value: amount)) {
+                return amount.trimmingCharacters(in: .whitespaces)
+            }
+        }
+        return "\(amount)"
+    }
+    
+    /// Get decimal amount from formatted currency code
+    /// - Parameters:
+    ///   - amount: amount string
+    ///   - currencyCode: currency code
+    /// - Returns: a decimal amount 
+    public static func getDecimalAmount(amount: String, currencyCode: String?) -> Double {
+        let doubleAmount = NSString(string: amount).doubleValue
+        if let currencyCode = currencyCode,
+            let currency = getCurrency(for: currencyCode) {
+            var fractionalDenominator: Double = 10
+            if currency.decimals > 0 {
+                for _ in 1..<currency.decimals {
+                    fractionalDenominator *= 10
+                }
+                return doubleAmount / fractionalDenominator
+            }
+        }
+        return doubleAmount
+    }
+
+    /// Format amount for currency code using users locale
+    /// - Parameter currencyCode: currency code
+    /// - Returns: a formatted amount string
+    public static func formatStringAmount(_ amount: String, with currencyCode: String) -> String {
+        if let transferAmountCurrency = getCurrency(for: currencyCode) {
```

Comment:
same as above

Reply:
Renamed and Removed String extension completely